### PR TITLE
feat: [FileCollection] add function to reatain multiple patterns

### PR DESCRIPTION
**Pull request by christianberkman**
**Description**
Supersedes #8953

Add a method to ``FileCollection`` that accepts an array of strings to retain multiple patterns in the file collection.

**Checklist:**
- [X] Securely signed commits
- [X] Component(s) with PHPDoc blocks, only if necessary or adds value
- [X] Unit testing, with >80% coverage
- [X] User guide updated
- [X] Conforms to style guide

### CODE REVIEW

File: user_guide_src/source/changelogs/v4.6.0.rst [hunk 1]
```diff
@@ -108,6 +108,8 @@ Model
 Libraries
 =========
 
+- Added `retainMultiplePatterns()` to `FileCollection` class.
```
Comment (kenjis) [inline user_guide_src/source/changelogs/v4.6.0.rst:111]: Can you add a link to the section `retainMultiplePatterns` in the page `file_collections.rst`?
See https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/documentation.rst#to-a-section

File: user_guide_src/source/changelogs/v4.6.0.rst [hunk 2]
```diff
@@ -108,6 +108,8 @@ Model
 Libraries
 =========
 
+- Added `retainMultiplePatterns()` to `FileCollection` class. See :ref:`FileCollection::retainMultiplePatterns() <file-collections-retain-multiple-patterns>`
```
Comment (kenjis) [inline user_guide_src/source/changelogs/v4.6.0.rst:111]:
```suggestion
- Added `retainMultiplePatterns()` to `FileCollection` class. See :ref:`FileCollection::retainMultiplePatterns() <file-collections-retain-multiple-patterns>`.
```

File: user_guide_src/source/libraries/file_collections.rst [hunk 3]
```diff
@@ -118,6 +118,18 @@ Examples:
 
 .. literalinclude:: files/015.php
 
+
+.. _file-collections-retain-multiple-patterns
```
Comment (kenjis) [inline user_guide_src/source/libraries/file_collections.rst:122]:
```suggestion
.. _file-collections-retain-multiple-patterns:
```

**Review by kenjis — APPROVED** [review]
LGTM! Thank you.

**Comment by christianberkman** [comment]
> LGTM! Thank you.

Thanks. I assume adding a similar function for `removePattern()`  would just be for convenience sake since `removePattern()` can be called consecutively to remove multiple patterns, opposed to calling `retainPattern` consecutively.

File: user_guide_src/source/libraries/files/016.php [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<?php
+
+$files->retainMultiplePatterns(['*.css*', '*.js']); // Would keep only *.css and *.js files
```
Comment (michalsn) [inline user_guide_src/source/libraries/files/016.php:3]:
```suggestion
$files->retainMultiplePatterns(['*.css', '*.js']); // Would keep only *.css and *.js files
```
Either the code or the comment is incorrect.

Comment (christianberkman) [reply user_guide_src/source/libraries/files/016.php:3]: That indeed is a typo. Thank you for catching it.

**Review by kenjis — APPROVED** [review]

**Review by michalsn — APPROVED** [review]

**Pull request merged**